### PR TITLE
code with poll components

### PR DESCRIPTION
**Pull request by kshashank22**

### CODE REVIEW

File: src/redux/reducers/pollSlice.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        state.list = action.payload.data;
+      })
+      .addCase(fetchedData.rejected, (state, action) => {
+        state.status = "failed";
+        state.error = action.error.message;
+      });
+  },
+});
+
+export default pollSlice.reducer;
```
Comment (akash-excell) [inline src/redux/reducers/pollSlice.jsx]: coding consistency is important.
you should keep code quality same.
previous slice and this slices is very different.
keep things consistent 
don't use chatgpt or any code writer ai

File: src/components/login/LogIn.jsx [hunk 2]
```diff
@@ -1,14 +1,30 @@
...
 import Button from "../button/Button";
+import { jwtDecode } from "jwt-decode";
 
 function LogIn() {
+  const loginSlice = useSelector((state) => state.loginSlice);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (loginSlice.isSuccess && loginSlice.data.token) {
+      const decoded = jwtDecode(loginSlice.data.token);
```
Comment (akash-excell) [inline src/components/login/LogIn.jsx:21]: save role in localStorage , you have to check again if user refresh the page

File: src/components/addoptionpoll/AddOptionPoll.jsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  const loading = useSelector((state) => state.optionsSlice.isLoading);
+  const status = useSelector((state) => state.optionsSlice.isSuccess);
+  const navigate = useNavigate();
+  const { addoptionId } = useParams();
+  const location = useLocation();
+
+  if (status) {
+    dispatch(resetReducer());
+    navigate("/adminpoll");
+  }
```
Comment (akash-excell) [inline src/components/addoptionpoll/AddOptionPoll.jsx]: again direct condition in component without hooks

File: src/components/addpoll/AddPoll.jsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+  const newOptionsList = [];
+
+  if (status) {
+    setNewOptions([{ option: "" }]);
+    setNewTitle("");
+    setAddNewPoll(!addNewPoll);
+    dispatch(fetchedData());
+    dispatch(resetReducer());
+    navigate("/adminpoll");
+  }
```
Comment (akash-excell) [inline src/components/addpoll/AddPoll.jsx]: ??
what is use of useEffect?

File: src/components/addpoll/AddPoll.jsx [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+  const handleClick = () => {
+    if (newOptions.length > 3) {
+      setError(true);
+    } else {
+      setNewOptions([...newOptions, { option: "" }]);
+    }
+  };
+
+  const newOptionsList = [];
```
Comment (akash-excell) [inline src/components/addpoll/AddPoll.jsx]: declare in useState

File: src/components/datalists/DataLists.jsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  if (optionSuccess) {
+    dispatch(fetchedData());
+    dispatch(resetReducer());
+  }
+
+  if (pollSuccess) {
+    dispatch(fetchedData());
+    dispatch(resetReducers());
+  }
+
```
Comment (akash-excell) [inline src/components/datalists/DataLists.jsx]: multiples??
what is use of useEffect then if you are calling condition directly

File: src/components/editpoll/EditPoll.jsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  const status = useSelector((state) => state.optionsSlice.isSuccess);
+  const navigate = useNavigate();
+  const { edittitleId } = useParams();
+  const location = useLocation();
+
+  if (status) {
+    dispatch(resetReducer());
+    navigate("/adminpoll");
+  }
+
```
Comment (kishankumar0508) [inline src/components/editpoll/EditPoll.jsx:23]: ??

Comment (akash-excell) [inline src/components/editpoll/EditPoll.jsx]: ????? again

**Pull request merged**